operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResource.java

```diff
@@ -12,4 +13,46 @@ default EventSource initEventSource(EventSourceContext<P> context) {
   default Class<R> resourceType() {
     return (Class<R>) Utils.getFirstTypeArgumentFromInterface(getClass());
   }
+
+  /**
+   * todo: return Optional<R></R> instead?
+   * 
+   * Computes the desired state of the dependent based on the state provided by the specified
+   * primary resource.
+   * 
+   * The default implementation returns {@code null} which corresponds to the case where the
+   * associated dependent should never be created by the associated reconciler or that the global
+   * state of the cluster doesn't allow for the resource to be created at this point.
+   * 
+   * @param primary the primary resource associated with the reconciliation process
+   * @param context the {@link Context} associated with the reconciliation process
+   * @return an instance of the dependent resource matching the desired state specified by the
+   *         primary resource or {@code null} if the dependent shouldn't be created at this point
+   *         (or ever)
+   */
+  default R desired(P primary, Context context) {
```

Comment:
I think we will need here an extended context, with the other resources included. If for exmample a resource is depending on an other resource output?

Reply:
That's what the context is for. This gives you accès to other resources.

Reply:
Was thinking a more explicit access, but for now this might be enough.